pkg/runtime/framework/plugins/volcano/volcano_test.go

```diff
@@ -0,0 +1,399 @@
+					PodGroupPolicy: &trainer.PodGroupPolicy{
+						PodGroupPolicySource: trainer.PodGroupPolicySource{
+							Volcano: &trainer.VolcanoPodGroupPolicySource{
+								NetworkTopology: &volcanov1beta1.NetworkTopologySpec{
+									Mode:               volcanov1beta1.HardNetworkTopologyMode,
+									HighestTierAllowed: ptr.To(1),
+								},
+							},
+						},
+					},
+				},
+				Scheduler: &runtime.Scheduler{},
+			},
+			existingPG: &volcanov1beta1.PodGroup{ObjectMeta: metav1.ObjectMeta{Name: "job-update", Namespace: "test-ns"}},
+			expectPG: volcanov1beta1ac.PodGroup("job-update", "test-ns").
+				WithOwnerReferences(metav1ac.OwnerReference().
+					WithAPIVersion(trainer.GroupVersion.String()).
+					WithKind(trainer.TrainJobKind).
+					WithName("job-update").
+					WithUID(types.UID(strconv.Itoa(2))).
+					WithController(true).
+					WithBlockOwnerDeletion(true)).
+				WithSpec(volcanov1beta1ac.PodGroupSpec().
+					WithMinMember(5).
+					WithMinResources(corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2300m"),
+						corev1.ResourceMemory: resource.MustParse("3Gi"),
+					}).
+					WithQueue("q1").
+					WithPriorityClassName("high-priority").
+					WithNetworkTopology(&volcanov1beta1ac.NetworkTopologySpecApplyConfiguration{
+						Mode:               ptr.To(volcanov1beta1.HardNetworkTopologyMode),
+						HighestTierAllowed: ptr.To(1),
+					})),
+			expectErr: nil,
+		},
+		"Error when getting existing PodGroup": {
+			trainJob: &trainer.TrainJob{
+				ObjectMeta: metav1.ObjectMeta{Name: "job-error", Namespace: "test-ns", UID: "3"},
+				Spec:       trainer.TrainJobSpec{Suspend: ptr.To(false)},
+			},
+			info: &runtime.Info{
+				TemplateSpec: baseInfo().TemplateSpec,
+				RuntimePolicy: runtime.RuntimePolicy{
+					PodGroupPolicy: &trainer.PodGroupPolicy{
+						PodGroupPolicySource: trainer.PodGroupPolicySource{
+							Volcano: &trainer.VolcanoPodGroupPolicySource{},
+						},
+					},
+				},
+				Scheduler: &runtime.Scheduler{},
+			},
+			existingPG: nil,
+			expectPG:   nil,
+			expectErr:  mockGetError,
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			clientBuilder := fake.NewClientBuilder().WithScheme(scheme)
+			if c.existingPG != nil {
+				clientBuilder.WithObjects(c.existingPG)
+			}
+			if name == "Error when getting existing PodGroup" {
+				clientBuilder.WithInterceptorFuncs(interceptor.Funcs{
+					Get: func(ctx context.Context, c client.WithWatch, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+						return mockGetError
+					},
+				})
+			}
+			cli := clientBuilder.Build()
+
+			v := &Volcano{
+				client: cli,
+				scheme: scheme,
+			}
+
+			_ = v.EnforcePodGroupPolicy(c.info, c.trainJob)
+			if c.expectInfo != nil {
+				if diff := cmp.Diff(c.expectInfo.Scheduler.PodAnnotations, c.info.Scheduler.PodAnnotations); diff != "" {
+					t.Errorf("PodAnnotations mismatch (-want +got):\n%s", diff)
+				}
+			}
```

Comment:
```suggestion
			if diff := cmp.Diff(c.expectInfo.Scheduler.PodAnnotations, c.info.Scheduler.PodAnnotations); diff != "" {
				t.Errorf("PodAnnotations mismatch (-want +got):\n%s", diff)
			}
```

We should always check the expected and actual object.